docs/conf.py

```diff
@@ -16,6 +16,15 @@
 sys.path.insert(0, os.path.abspath(".."))
 sys.path.append(os.path.abspath("./_ext"))
 
+# -- RTD setup --------------------------------------------------------------
+
+#RZ, 20200508: If protobufs don't exist, build them
+#Catching ImportError on Alert_pb2 assumes Alert continues to be part of st lib
+try:
+    from streamlit.proto import Alert_pb2
+except ImportError:
+    import subprocess
+    subprocess.run(["bash", "env_setup.sh"])
```

Comment:
Is `env_setup` strictly necessary here? 

That is, is it possible to instead tell the user "please run `make all-devel` (or `make protobuf`) from the Streamlit root directory" and error out, or is this a script that's supposed to run without user intervention?

(Alternately, could we add `make readthedocs` to our Makefile, to avoid having to run the env_setup script with its pip install shenanigans?)

Reply:
I did this to avoid having to do a much larger setup. We don't need to do the whole build process for the docs to build correctly, and RTD also has issues with installing random C dependencies. Which is to say, I can try, but this seems much more efficient that setting up the entire dev environment.

Reply:
The pip install bit is to control the order of operations. RTD will install the streamlit package first from source, but that doesn't have the protobufs. So then I build them, and reinstall. Not sure our make process can really handle that, since what RTD is doing is opaque to our normal process.

Reply:
Upon more thought, not sure the `make` process can work, because we can't assume that we can modify the RTD Docker container with these updates: https://github.com/streamlit/streamlit/wiki/Contributing#ubuntu

Unless I'm missing something, sudo access isn't available, which is why my attempt at using apt didn't work here https://github.com/streamlit/streamlit/pull/1435/commits/a9831e12a4a1274885c4a5a82a282762efac4e0b

Reply:
That makes good sense - could you add some inline docs explaining why the script duplicates some Makefile logic, for future visitors to the code?

Reply:
+1